File: src/main/java/ru/sfedu/teamselection/domain/Student.java
package ru.sfedu.teamselection.domain;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.Size;
import java.util.ArrayList;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import ru.sfedu.teamselection.domain.application.Application;

/**
 * Entity for students
 */
@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "students")
public class Student {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private Integer course;

    @Column(name = "group_number")
    private Integer groupNumber;

    @Column(name = "about_self")
    @Size(max = 1024)
    private String aboutSelf;

    @Column
    @Size(max = 255)
    private String contacts;

    @Column(name = "has_team", nullable = false)
    @Builder.Default
    private Boolean hasTeam = false;

    @ManyToOne(fetch = FetchType.LAZY,
            cascade = {CascadeType.PERSIST, CascadeType.MERGE})
    private Team currentTeam;

    @Column
    @ManyToMany
    @JoinTable(
            name = "teams_students",
            joinColumns = @JoinColumn(name = "team_id"),
            inverseJoinColumns = @JoinColumn(name = "student_id")
    )
    @Builder.Default
    private List<Team> teams = new ArrayList<>();

    @Column(name = "is_captain", nullable = false)
    @Builder.Default
    private Boolean isCaptain = false;


    @ManyToOne(fetch = FetchType.EAGER)
    private Track currentTrack;

    @Column(name="hasStudentDebt")
    @Builder.Default
    private Boolean hasStudentDebt = false;

    @Column
    @ManyToMany
    @JoinTable(
            name = "students_technologies",
            joinColumns = @JoinColumn(name = "student_id"),
            inverseJoinColumns = @JoinColumn(name = "technology_id")
    )
    @Builder.Default
    private List<Technology> technologies = new ArrayList<>();

    @Column
    @OneToMany(mappedBy = "student")
    @Builder.Default
    private List<Application> applications = new ArrayList<>();

    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
}

File: src/main/kotlin/ru/sfedu/teamselection/domain/Student.kt
package ru.sfedu.teamselection.domain

import com.fasterxml.jackson.annotation.JsonBackReference
import com.fasterxml.jackson.annotation.JsonProperty
import org.springframework.boot.context.properties.bind.DefaultValue
import ru.sfedu.teamselection.enums.Roles
import jakarta.persistence.*

@Entity
@Table(name = "students")
@Access(AccessType.FIELD)
class Student(
    @Column
    @JsonProperty
    var fio: String? = null,
    @Column
    @JsonProperty
    var email: String? = null,
    @Column
    @JsonProperty
    var course: Int? = null,
    @Column
    @JsonProperty
    var groupNumber: Int? = null,
    @Column
    @JsonProperty
    var aboutSelf: String? = null,
    @Column
    @JsonProperty
    var tags: String? = null, // Строка разделенная пробелами
    @Column
    @JsonProperty
    var contacts: String? = null,
    @Column
    @JsonProperty
    @DefaultValue(value = ["false"])
    var status: Boolean? = null, // статус участик команды/не участник
    @ManyToOne(fetch = FetchType.LAZY)
    @JsonBackReference
    var currentTeam: Team? = null,
    @Column
    var trackId: Long? = null,
    @Column
    @DefaultValue(value = ["false"])
    @JsonProperty
    var captain: Boolean? = null, //TODO поменять логику начального значения
    @Column
    @JsonProperty
    var subscriptions: String? = "",
    @OneToOne
    @JsonProperty
    var user: Users? = null
): Domain() {
    override fun toString(): String { //TODO fix
        return ""
    }

    override fun equals(other: Any?): Boolean {
        return true
    }
}